kvbc/src/reconfiguration_kvbc_handler.cpp

```diff
@@ -823,6 +823,23 @@ bool ReconfigurationHandler::handle(const concord::messages::PruneRequest& comma
   return true;
 }
 
+// This is the only way to trigger the compaction outside of a snapshot if the system is idle
+bool ReconfigurationHandler::handle(const concord::messages::PruneCompactRequest& command,
```

Comment:
@cloudnoize if I don't create this handler, the `KeyValueBlockchain::onNewBFTSequenceNumber()` method is not called and therefore the check for compaction is not done.

Is there another less intrusive way to trigger this method ?

Reply:
```onNewBFTSequenceNumber``` currently is called only when a new block is added with a new BFT sequence number.
We can probably add a call to this method as a callback from the BFT engine after it has finished the execution of requests. It will remove the dependency on a new subsequence request to release the snapshot  and will support the required behavior I think

Reply:
Yes, this is an optimization: we could call the compaction immediately after the execution of PruneRequest once the reconfiguration block is persisted.